doc/ir-semantics.text

```diff
@@ -120,9 +120,9 @@ Update (partial) function, returns an H:
     where v := eval P H E e
   osr-env P H E M (mut x)       = (x ↦ ⊥)
   osr-env P H E M (mut x = &y)  = (x ↦ ref a)
-    where a = E(y)
-  osr-env P H E M (mut x = e)   = (x ↦ ref e)
-    where (a ↦ _) = M(x)
+    where E ∋ (y ↦ ref a)
+  osr-env P H E M (mut x = e)   = (x ↦ ref a)
+    where M ∋ (x ↦ (a ↦ _))
```

Comment:
That's what I thought at first, but `materialize` does the evaluation of `e` and updates the heap. So the result is already in `M` before `osr-env` is called. (And `osr-env` only updates the environment.)

On one hand, it makes sense to keep the two separate, On the other hand, it might be clearer to update both the heap and environment in the same place.